### PR TITLE
fix: Avoid destructuring `connectionData` when value is undefined

### PR DESCRIPTION
**Pull request by KevLehman**
## Issue(s)

SUP-672
## Further comments

#32620 introduced the bug, mainly for apps engine, which caused that when GDPR setting is enabled, new visitors cannot be created. This is because the server was trying to destructure `connectionData`, which was undefined.

The code on `LivechatTyped` was edited in a way that it was not checking `connectionData` to be an object before destructuring, causing the visitor creation to error out before completion.

### CODE REVIEW

File: .changeset/brave-brooms-invent.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@rocket.chat/meteor": patch
+---
+
+Fixed a problem that caused visitor creation to fail when GDPR setting was enabled and visitor was created via Apps Engine or the deprecated `livechat:registerGuest` method.
```
Comment (matheusbsilva137) [inline .changeset/brave-brooms-invent.md:5]:
```suggestion
Fixes a problem that caused visitor creation to fail when GDPR setting was enabled and visitor was created via Apps Engine or the deprecated `livechat:registerGuest` method.
```
Not an issue, but I found out we changed our changeset best practices docs and we should now use the present tense. I got surprised by that, just found an opportunity to spread the word :kekw:

**Review by matheusbsilva137 — DISMISSED** [review]

File: .changeset/brave-brooms-invent.md (hunk 1 above)
Comment (KevLehman) [reply .changeset/brave-brooms-invent.md:5]: Well

**Review by aleksandernsilva — APPROVED** [review]

**Pull request merged**